homeassistant/components/calendar/__init__.py

```diff
@@ -216,7 +212,16 @@ async def get(self, request: web.Request, entity_id: str) -> web.Response:
         event_list = await entity.async_get_events(
             request.app["hass"], start_date, end_date
         )
-        return self.json(event_list)
+        return self.json(
+            [
+                {
+                    "summary": event.summary,
+                    "start": _get_api_date(event.start),
+                    "end": _get_api_date(event.end),
```

Comment:
How are all day events identified here?

Reply:
Purely based on the string format being a date (e.g. `2022-03-28`).
You can see here that all date strings are treated the same and that the all day event field is ignored:
https://sourcegraph.com/github.com/home-assistant/frontend/-/blob/src/data/calendar.ts?L44

This is what the API response looks like:
```
 {"summary": "#Backyard All Day Event", "start": {"date": "2022-03-28"}, "end": {"date": "2022-03-29"}}
```
And this is how its rendered by the calendar widge:

<img width="945" alt="Screen Shot 2022-03-29 at 9 13 54 PM" src="https://user-images.githubusercontent.com/6026418/160750262-dbddead4-db01-4d67-bf04-743d254360e7.png">

My assumption here is that we only care about rendering in the frontend for this API.